### PR TITLE
CR-3510: Added sidebar

### PR DESCRIPTION
**Pull request by Tomerva**

### CODE REVIEW

File: src/commands/scan.ts [hunk 1]
```diff
@@ -36,6 +31,14 @@ export async function scan(diagnostics: vscode.DiagnosticCollection) {
         }, async(progress) => {
             scanInProgress = true;
             progress.report({ increment: 0, message: 'Starting Cloudrail run'});
+
+            diagnostics.clear();
+            let runResults: CloudrailRunResponse | undefined;
+            const config = await getConfig();
+            const terraformWorkingDirectory = await getTerraformWorkingDirectory();
+            if (!terraformWorkingDirectory) {
+                return;
```
Comment (alonashkenazi) [inline src/commands/scan.ts:40]: Should we show/log an informative message?

File: src/commands/scan.ts [hunk 2]
```diff
@@ -63,17 +66,21 @@ export async function scan(diagnostics: vscode.DiagnosticCollection) {
...
             progress.report({ increment: 100, message: 'Applying scan results...'});
-            await handleRunResults(runResults!, diagnostics, terraformWorkingDirectory);
+            await handleRunResults(runResults!, diagnostics, terraformWorkingDirectory, sidebarProvider);
+            isSuccessfulScan = true;
         });
     } catch(e) {
         logger.error(`Failed to perform scan. reason: ${e}`);
         vscode.window.showErrorMessage(`An unknown error has occured while performing the scan. Check log for more information: ${logPath}`);
     } finally {
+        if (!isSuccessfulScan) {
```
Comment (alonashkenazi) [inline src/commands/scan.ts:76]: Should we also show a message to the user?

File: src/sidebar/cloudrail_issue_info_provider.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            this.view.html = `No issue selected. Please run Cloudrail Scan and choose an issue`;
+        }
+    }
+
+    private toHtml(element: CloudrailTreeItem): string {
+        if (element instanceof CloudrailRuleTreeItem) {
+            return `
+            <h2>${element.ruleName}</h2>
+
+            <h3>Severity: ${element.severity}</h3>
```
Comment (alonashkenazi) [inline src/sidebar/cloudrail_issue_info_provider.ts:32]: only Severity should have h3, same connect for the other headers

**Review by alonashkenazi — APPROVED** [review]

File: src/commands/scan.ts (hunk 1 above)
Comment (Tomerva) [reply src/commands/scan.ts:40]: we do it as part of getTerraformWorkingDirectory

File: src/commands/scan.ts (hunk 2 above)
Comment (Tomerva) [reply src/commands/scan.ts:76]: we show a message in the issue explorer (line below)
also if there was a specific error we would of already displayed it -
when no valid terraform directory specified
when a mandatory field is not set
when cloudrail failed for some reason

**Pull request merged**